src/libPersistence/BlockStorage.cpp

```diff
@@ -628,15 +724,20 @@ bool BlockStorage::GetDSCommittee(shared_ptr<DequeOfNode>& dsCommittee,
   LOG_MARKER();
 
   unsigned int index = 0;
-  lock_guard<mutex> g(m_mutexDsCommittee);
+  shared_lock<shared_timed_mutex> g(m_mutexDsCommittee);
   string strConsensusLeaderID = m_dsCommitteeDB->Lookup(index++);
```

Comment:
Maybe let `m_mutexDsCommittee` to protect `string strConsensusLeaderID = m_dsCommitteeDB->Lookup(index++);` only

Reply:
Nope... notice that there is a while loop below